Add Nether Star Nuggets to Compacting Drawers

ebmusicman84 opened the pull request.

ebmusicman84 committed:
Add Nether Star Nuggets to Compacting Drawers

eutropius225 commented:
Nether star nuggets cannot currently be converted back into nether stars, so this would add the currently non-existent conversion from nuggets to stars.

ebmusicman84 commented:
I'd say that's not a problem as there's currently no way to make Nether Star Nuggets without "uncrafting" a Nether Star, so this doesn't create a loop or anything.

ALongStringOfNumbers commented:
You can technically get them back lossily by pulverizing the nuggets into small dust, combining into regular dust, and then running through the implosion compressor at a 4 dust to 3 nether star ratio. However, I still would be in favor of the ability to recombine the nuggets into nether stars.

eutropius225 commented:
I'm fine with these changes, just add a crafting recipe conversion as well at that point.

ALongStringOfNumbers commented:
Agreed, might as well add a crafting table recipe

ebmusicman84 committed:
Nuggets -> Nether Star Crafting Table

As discussed in the PR.

ebmusicman84 commented:
Did so @ALongStringOfNumbers

Exaxxion commented:
This sounds like a reasonable quality of life improvement.

Since the whole grid is filled with the same item, the recipe for combining could be implemented more simply as a shapeless one, like the Luminessence one above it. Unlike that recipe though, we should name it as that's good practice (otherwise it gets a randomly generated name which doesn't really help if you're trying to look through the scripts).

ebmusicman84 committed:
Update ExtendedCrafting.zs

Switched to shapeless as requested.

I tried doing <extendedcrafting:material:140>*9 but it only allowed a stack of nine in one slot, instead of one in nine slots.  

I'm super n00b on CT though so if I'm just ignorant here please correct me.

Exaxxion commented:
I'm sorry, I realize that what I said was rather confusing. I need to make sure I finish my morning coffee before reviewing pull requests.

The other recipe I mentioned is actually the opposite direction of what you're trying to do. What I should have said is you can alternately use the `makeShapeless3` function from `CommonVars.zs`: the syntax is the same as `makeRecipe` but it generates a shapeless recipe rather than a shaped one.

You can revert your latest commit and it should be fine, shapeless really doesn't matter when the whole grid is filled anyway.

ebmusicman84 committed:
Revert Previous Change

I should just install the desktop app lol
